src/AppBundle/Geocoder/EntityAddressGeocodingSubscriber.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+namespace AppBundle\Geocoder;
+
+use AppBundle\Membership\AdherentAccountSelfActivatedEvent;
+use AppBundle\Membership\AdherentEvents;
+use Doctrine\Common\Persistence\ObjectManager;
+use Geocoder\Geocoder;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+
+class EntityAddressGeocodingSubscriber implements EventSubscriberInterface
+{
+    private $geocoder;
+    private $manager;
+
+    public function __construct(Geocoder $geocoder, ObjectManager $manager)
+    {
+        $this->geocoder = $geocoder;
+        $this->manager = $manager;
+    }
+
+    public function onAdherentAccountActivationCompleted(AdherentAccountSelfActivatedEvent $event)
+    {
+        $this->updateGeocodableEntity($event->getAdherent());
+    }
+
+    private function updateGeocodableEntity(GeocodableInterface $geocodable)
+    {
+        if ($coordinates = $this->geocode($geocodable->getGeocodableAddress())) {
+            $geocodable->updateCoordinates($coordinates);
+            $this->manager->flush();
+        }
+    }
+
+    private function geocode(string $address)
+    {
+        try {
+            $addresses = $this->geocoder->geocode($address);
+        } catch (\Exception $e) {
+            // do nothing when an exception arises
+            return null;
+        }
+
+        if (count($addresses)) {
+            return Coordinates::createFromAddress($addresses->first());
+        }
+    }
+
+    public static function getSubscribedEvents()
+    {
+        return [
+            AdherentEvents::ACTIVATION_COMPLETED => ['onAdherentAccountActivationCompleted', -256],
```

Comment:
I think we should not authorize the subscription when the address is not valid. Can I add this in the US?